rfcs/2022-03-22-motion-rfc.md

```diff
@@ -0,0 +1,611 @@
+                    outputRange: [0, -50, 50, -50, 50, -50, 50, -50, 50, -50, 0],
+                    easing: motionToken.easing.standard.effective, // Define easing inside the interpolate function
+                  }),
+                },
+              ],
+            },
+          ]}
+        />
+        <Button text="Shake" onPress={()=>
+          Animated.timing(shakeAnimation, {
+            toValue: 100,
+            useNativeDriver: false, // Set native driver usage to false
+            duration: motionToken.duration.duration3, // Define only duration here
+          }).start()}
+        />
+      </View>
+    );
+  }
+  ```
+- You will need to ensure that you set `useNativeDriver` to `false` since adding easing to `interpolation` is not supported with `useNativeDriver`
+
+# Alternatives
+### Spring Animations instead of Easing Animations
+- Instead of using Easing for our animations we can use spring-physics based animations. 
+- Spring animations *claim* to look more fluid and natural since the animations are based on the same physical rules as that of a spring. 
+- Spring animations can be controlled by defining the characteristics of a spring like `Mass`, `Tension` & `Friction`
+- To use spring animations, we need to use some 3rd party library like `framer-motion` or `react-spring` since it is not natively supported by CSS.
+- **Non-realtime** motion include a bunch of actions like scale, move, fade, fill, etc. All of which can be achieved using **spring animations as well as easing animations**.
+- **Realtime** motion is a lot more complex and we would **not be able to achieve** the same fluid animations **just** by **using easing** animations.
+- Spring animations make the most sense for animations that are dependent on user interactions like dragging chat bubbles, dragging bottom sheet to open and close, swiping cards to dismiss them, swiping to another tab
+- **We will dive deeper** into whether or not we want to use **spring animations** when we explore **realtime motion** **in the future**. As of the scope of this RFC which is only Non-realtime motion, we will continue using easing animations.
+- You can read more about spring based animations here:
+  - [react-spring.io](https://react-spring.io/)
+  - [A friendly introduction to spring physics
+](https://www.joshwcomeau.com/animation/a-friendly-introduction-to-spring-physics/)
+  - [The physics behind spring animations
+](https://blog.maximeheckel.com/posts/the-physics-behind-spring-animations/) 
+
+# Adoption strategy
+- Consumers can start using motion tokens as they would with other DS tokens.
+- Consumers would need to be aware of the drawback mentioned above at [Library specific tokens](#library-specific-tokens) before they start using them
+- Blade's developers can use this RFC as a guideline to add motion to their components
+
+# How do we educate people?
+- Have a detailed guideline on our documentation site
+
+# Open Questions
```

Comment:
Should we also include a section on how to handle user preference of `prefers-reduced-motion` to stop these motion animations?

Reply:
Good question. Not sure where this fits in right now. 

1. Would consumers take care of implementing alternative "less animated" versions using different tokens that we'll expose?
2. Would we change the values of the tokens depending on `prefers-reduced-motion`? 
3. Would we need to change motion for the components we expose from blade if we detect `prefers-reduced-motion`? 
4. Should we figure these out when we explore with a11y?

Let me discuss this with the rest of the team and get back to you.

// @kamleshchandnani @saurav12